server/consumer.go

```diff
@@ -603,6 +608,16 @@ func (o *Consumer) setLeader(isLeader bool) {
 		// Restore our saved state. During non-leader status we just update our underlying store.
 		o.readStoredState()
 
+		// Do info sub.
+		if o.infoSub == nil && mset != nil && mset.jsa != nil {
+			mset.mu.RLock()
```

Comment:
I don't know enough of the locking rules between stream and consumer, but here it seem we have o.mu.Lock() above and now trying to get mset's lock. Could that cause a lock inversion (again without knowing, I would think that it seem more logical to have stream -> consumer than the opposite).

(if that is an issue, would move those mset's fields capture before getting o.mu.Lock(), even if we end-up not needing them).

Reply:
Good point, will cleanup.